lib/api/objectPut.js

```diff
@@ -103,7 +103,10 @@ function objectPut(authInfo, request, streamingV4Params, log, callback) {
                     'putObject');
                 return callback(err, responseHeaders);
             }
-            const newByteLength = request.parsedContentLength;
+            const isIngest = request.headers['x-amz-meta-mdonly'] || false;
```

Comment:
does it have an adverse effect if a regular S3 client uses these user metadata headers on a put object requests?

Reply:
If a regular S3 client uses these headers and `x-amz-meta-size` is undefined, then it would cause `newByteLength` to be undefined. I can add a condition to check if the size header is defined.

Reply:
oh! that's a good one! I didn't see it. My original thought was about the ability a regular S3 client can have to mess with the prom metrics

Reply:
Ah, I see your point. Yes, there is a potential to mess with the ingestion metrics if someone were to use both the `size` and `mdonly` headers in the PUT object. Fortunately, these metrics are there to simply provide estimates and are not generally valued as the "source of truth"

Reply:
The same would apply to the UTAPI metrics which is used for billing purposes. Those metrics need to be precise and probably shouldn't be alterable via user-metadata.

Reply:
particularly if you can audit the code for such things

Reply:
Looks like we could have an easy win by keeping the `newByteLength` value sent to UTAPI different from that needed for the prometheus metric. That way we'd for sure avoid any potential issue with that.

Reply:
Prometheus isn't used for billings, so I'll update the code to not modify the `newByteLength`.

Reply:
Yeah exactly.